### PR TITLE
Rule: `pointless-reassignment`

### LINKED ISSUE #867: Rule: `pointless-reassignment`
Since values and variables are immutable in Rego, re-assigning a variable from one name to another doesn't come with any benefits.

**Avoid**
```rego
allow if {
    users := all_users
    # do something with users
}
```

Re-assigning a long ref to a variable should however be considered OK:

```rego
allow if {
    users := data.context.permissions.users
    # do something with users
}
```

### PR DESCRIPTION
**Pull request by anderseknert**
Also:
- fix the one violation of this rule found in our code :P
- add metadata annotations to a bunch of rules

Fixes #867

### CODE REVIEW

File: docs/rules/style/pointless-reassignment.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```rego
+package policy
+
+allow if {
+    users := input.context.permissions.users
+    any_admin(users)
+}
+```
+
+This rule does not consider that a violation.
```
Comment (charlieegan3) [inline docs/rules/style/pointless-reassignment.md:44]:
```suggestion
This rule does not consider such examples violations.
```

Comment (charlieegan3) [reply docs/rules/style/pointless-reassignment.md:44]: I wonder what if I a had a rule like `foo_bar_whizzz_bang_bar_foo_baz_v1_foo`, and then reassigned it for readability, would this be good to ignore too?

File: bundle/regal/rules/imports/circular_import.rego [hunk 2]
```diff
@@ -43,11 +43,10 @@ aggregate_report contains violation if {
 
 	sorted_group := sort(g)
 
-	location := [e |
+	location := [loc |
 		some m1 in sorted_group
 		some m2 in sorted_group
 		some loc in package_locations[m1][m2]
-		e := loc
```
Comment (charlieegan3) [inline bundle/regal/rules/imports/circular_import.rego:50]: my handiwork!

**Review by charlieegan3 — APPROVED** [review]

File: docs/rules/style/pointless-reassignment.md (hunk 1 above)
Comment (anderseknert) [reply docs/rules/style/pointless-reassignment.md:44]: I'd say the issue then would be you naming a rule or var like that in the first place :D

**Pull request merged**